src/views/view.c

```diff
@@ -1222,6 +1226,85 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
   return missing;
 }
 
+void
```

Comment:
Yet, I'll see tomorrow if I can put the common code in a specific procedure. I don't remember all the details as this was done long time ago now.